src/views/RequiredSignoffs/index.jsx

```diff
@@ -1,18 +1,31 @@
-import React from 'react';
-import { Switch } from 'react-router-dom';
+import React, { Fragment } from 'react';
+import { Switch, Redirect } from 'react-router-dom';
 import RouteWithProps from '../../components/RouteWithProps';
 import routes from './routes';
+import { withUser } from '../../utils/AuthContext';
 
-export default function Rules(props) {
+function RequiredSignoffs(props) {
   const {
+    user,
     match: { path },
   } = props;
 
   return (
-    <Switch>
-      {routes(path).map(({ routes, ...routeProps }) => (
-        <RouteWithProps key={routeProps.path || 'not-found'} {...routeProps} />
-      ))}
-    </Switch>
+    <Fragment>
+      {user ? (
+        <Switch>
+          {routes(path).map(({ routes, ...routeProps }) => (
+            <RouteWithProps
+              key={routeProps.path || 'not-found'}
+              {...routeProps}
+            />
+          ))}
+        </Switch>
+      ) : (
+        <Redirect to="/" />
+      )}
+    </Fragment>
```

Comment:
A view shouldn't be concerned with the redirect flow in my opinion. Instead of having to add the redirect logic in each protected view, what do you think about the following:

```diff
--- a/src/components/RouteWithProps/index.jsx
+++ b/src/components/RouteWithProps/index.jsx
@@ -1,5 +1,6 @@
 import React, { Suspense } from 'react';
-import { Route } from 'react-router-dom';
+import { Route, Redirect } from 'react-router-dom';
+import { withUser } from '../../utils/AuthContext';

 /**
  * Conditionally render a component based on location, with non-react-router
@@ -7,6 +8,8 @@ import { Route } from 'react-router-dom';
  */
 function RouteWithProps(props) {
   const {
+    user,
+    requiresAuth,
     component: Component,
     path,
     exact,
@@ -25,11 +28,15 @@ function RouteWithProps(props) {
       sensitive={sensitive}
       render={({ staticContext, ...renderProps }) => (
         <Suspense fallback={null}>
-          <Component {...renderProps} {...rest} />
+          {requiresAuth && !user ? (
+            <Redirect to="/" />
+          ) : (
+            <Component {...renderProps} {...rest} />
+          )}
         </Suspense>
       )}
     />
   );
 }

-export default RouteWithProps;
+export default withUser(RouteWithProps);
```

Then we could protect individual routes simply by adding `requiresAuth: true` in `route.js`.

_Example: Protecting /users, /roles, /required-signoffs_

```diff
--- a/src/routes.js
+++ b/src/routes.js
@@ -46,15 +46,18 @@ export default [
     component: Users,
     path: '/users',
     routes: userRoutes('/users'),
+    requiresAuth: true,
   },
   {
     component: Roles,
     path: '/roles',
+    requiresAuth: true,
   },
   {
     component: RequiredSignoffs,
     path: '/required-signoffs',
     routes: requiredSignoffsRoutes('/required-signoffs'),
+    requiresAuth: true,
   },
```

Reply:
Makes sense to me, let's do it :)